### PR TITLE
Add AUTO_LOWERCASE_KEYBOARD

### PR DESCRIPTION
**Pull request by pkmnsnfrn**
# Description
Keyboard changes from upper to lower after first character

* Adds `AUTO_LOWERCASE_KEYBOARD` which swaps the keyboard to the lowercase after the first capital letter

#### `SwapKeyboardToLowerAfterFirstCapitalLetter`
This function runs when the following are true:
* `AUTO_LOWERCASE_KEYBOARD` is enabled
* When the keyboard is set to capital
* The second character is being input

The keyboard will automatically swap to lowercase.

# Testing
## Clean Branch
You can recreate this branch by applying a patch or pulling the repo. From a clean version of expansion's upcoming, you can either:

### Patch
`wget https://files.catbox.moe/v0z1pc.patch -O keyboard.patch ; git apply keyboard.patch ; rm keyboard.patch`

### Repo
`git remote add psf-expansion https://github.com/PokemonSanFran/pokeemerald-expansion/ ; git pull psf-expansion swapKeyboard`

## Manual Tests
After replicating the branch, to recreate my testing environment, you can either directly download the debug script, or manually create the changes.

#### `TRUE`
`wget https://files.catbox.moe/y2knrw.h -O include/config/general.h`
#### `FALSE`
`wget https://files.catbox.moe/3y97pw.h -O include/config/general.h`

### Manual Testing
* Modify `include/config/general.h` to the desired values
* Open the Debug Menu, Utilities > Player Name
* Input a capital letter

## Verified Scenarios
All videos show:
* Open the Debug Menu, Utilities > Player Name
* Input a capital letter

### TRUE
https://github.com/user-attachments/assets/1327d1a6-82f7-45d4-b8db-12a8ff1adb82

### FALSE
https://github.com/user-attachments/assets/8d3c3c57-8100-47cf-9cd3-bbc10e006630

# **People who collaborated with me in this PR**
This was originally [written by](<https://github.com/pret/pokeemerald/wiki/Automatically-make-the-keyboard-switch-to-lowercase-after-the-first-character/>) by @Jaizu. Give them all the credit.

# Features this PR does NOT handle:

# Discord Contact Info

I am `pkmnsnfrn` on Discord.

### CODE REVIEW

File: src/naming_screen.c [hunk 1]
```diff
@@ -1480,6 +1481,8 @@ static bool8 KeyboardKeyHandler_Character(u8 input)
     {
         bool8 textFull = AddTextCharacter();
 
+		SwapKeyboardToLowerAfterFirstCapitalLetter();
```
[the remark this answers is not in the document]
Comment (pkmnsnfrn) [reply src/naming_screen.c:1484]: okay I swear rn I don't see a tab character in this file but let's see if this commit fixes it

[the remark this answers is not in the document]
Comment (pkmnsnfrn) [reply src/naming_screen.c]: https://github.com/rh-hideout/pokeemerald-expansion/pull/4976/commits/3c71f1112ec492d47577166d77fdcb957e66ab11

File: src/naming_screen.c [hunk 2]
```diff
@@ -1436,6 +1436,7 @@ static void NamingScreen_CreateWaldaDadIcon(void)
 //--------------------------------------------------
 
 static bool8 KeyboardKeyHandler_Character(u8);
+static void  SwapKeyboardToLowerAfterFirstCapitalLetter(void);
```
[the remark this answers is not in the document]
Comment (pkmnsnfrn) [reply src/naming_screen.c]: https://github.com/rh-hideout/pokeemerald-expansion/pull/4976/commits/3c71f1112ec492d47577166d77fdcb957e66ab11

File: include/config/general.h [hunk 3]
```diff
@@ -80,4 +80,7 @@
 #define UNITS                        UNITS_IMPERIAL
 #define CHAR_DEC_SEPARATOR           CHAR_PERIOD // CHAR_PERIOD is used as a decimal separator only in the UK and the US. The rest of the world uses CHAR_COMMA.
 
+// Naming Screen
+#define AUTO_LOWERCASE_KEYBOARD      GEN_LATEST  // Starting in GEN_6, after entering the first uppercase character, the keyboard switches to lowercase letters;
```
Comment (PCG06) [inline include/config/general.h:84]: More of a nitpick if anything, but there's a semi colon at the end of the comment, instead of a comma.

Comment (pkmnsnfrn) [reply include/config/general.h]: https://github.com/rh-hideout/pokeemerald-expansion/pull/4976/commits/888b0e0b44395e33a5b3ed9a00b96701ea4b1416

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**